packages/client/src/utils/utils.ts

```diff
@@ -131,7 +131,7 @@ export function convertPeerDescriptorToNetworkPeerDescriptor(descriptor: PeerDes
     }
     return {
         ...omit(descriptor, 'kademliaId'),
-        id: PeerID.fromValue(descriptor.kademliaId).toString(),
+        id: PeerID.fromValue(descriptor.kademliaId).toKey(),
```

Comment:
The original implementation which called `toString` _should_ be the correct one. When we want to get the string representation of a `PeerID` and there is an implemented method `PeerID#toString`, it make a lot of sense to use that method.

I'd suggest that we correct the `PeerID` class to work correctly. (Maybe the class doesn't even need two representations of the data, i.e. we could remove `PeerIDKey`?)

Reply:
Thoughts on this @juslesan 

Also is there a reason we need two different string representations?

Reply:
The key representation is absolutely there intentionally. For example the keys are used in map structures as the UTF8 string outputs have a danger colliding in cases where PeerIDs are generated from anything other than a string.

The safest way to ensure uniqueness of the PeerIDs is to use them as keys. If anything the toString representation should be removed, although it can be a useful tool in development

Reply:
So If I understood correctly @juslesan :
- toKey() returns a unique identifier
- toString() returns a human-readable identifier that is _usually_ unique but has the possibility of conflict?

Reply:
If possible it would be best to have only one string representation. E.g. if we see from a log that we receive a message from peer `foo` (`peerId.toString()`), but from a debugger we see that a message is in a map with key `bar` (`peerId.toKey()`), it is very misleading and slows down our debugging etc.